### PR TITLE
Feature/add bubble and selection sort

### PR DESCRIPTION
**Pull request by Sunbelife**
add bubble and selection sort

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/Llama-Group/Project-Llama/pull/60" target="_blank">ReviewNinja

**Comment by 0xsuu** [comment]
No need to create a new pr

File: llama_cpp/src/Sort/SelectionSort.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		SelectionSort() {}
+		SelectionSort(std::vector<T> *inputArray) { this->sort(inputArray); }
+
+		void doSort(std::vector<SortObject<T>> *reformedArray) override {
+			for (int i = 0; i < reformedArray->size(); i++) {
+				SortObject<T> key = reformedArray->at(i);
+				int flag = i;
+				// Select the min value
+				for (int j = i; j < reformedArray->size(); j++){
+					if (key > reformedArray->at(j)){
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:22]: Space

File: llama_cpp/src/Sort/SelectionSort.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	public:
+		SelectionSort() {}
+		SelectionSort(std::vector<T> *inputArray) { this->sort(inputArray); }
+
+		void doSort(std::vector<SortObject<T>> *reformedArray) override {
+			for (int i = 0; i < reformedArray->size(); i++) {
+				SortObject<T> key = reformedArray->at(i);
+				int flag = i;
+				// Select the min value
+				for (int j = i; j < reformedArray->size(); j++){
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:21]: Space

File: llama_cpp/src/Sort/SelectionSort.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+					if (key > reformedArray->at(j)){
+						flag = j;
+					}
+				}
+				// Swap between flag and the i if they are different
+				if (flag != i){
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:27]: Space

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 4]
```diff
@@ -43,28 +45,77 @@ static void BFSortDouble(benchmark::State& state) {
...
     std::vector<int> a = {5, 4, 3, 2, 1};
     while (state.KeepRunning()) {
-        inSort.sort(&a);
+        Insort.sort(&a);
     }
 }
 
-static void InSortString(benchmark::State& state) {
-    llama::InsertionSort<std::string> inSort = llama::InsertionSort<std::string>();
+static void InsortString(benchmark::State& state) {
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:57]: InSort

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 5]
```diff
@@ -43,28 +45,77 @@ static void BFSortDouble(benchmark::State& state) {
...
+    llama::InsertionSort<std::string> Insort = llama::InsertionSort<std::string>();
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:58]: InSort

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 6]
```diff
@@ -43,28 +45,77 @@ static void BFSortDouble(benchmark::State& state) {
...
     std::vector<std::string> a = {"5", "4", "3", "2", "1"};
     while (state.KeepRunning()) {
-        inSort.sort(&a);
+        Insort.sort(&a);
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:61]: InSort

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 7]
```diff
@@ -43,28 +45,77 @@ static void BFSortDouble(benchmark::State& state) {
...
 }
 
-static void InSortDouble(benchmark::State& state) {
-    llama::InsertionSort<double> inSort = llama::InsertionSort<double>();
+static void InsortDouble(benchmark::State& state) {
+    llama::InsertionSort<double> Insort = llama::InsertionSort<double>();
     std::vector<double> a = {5.0, 4.0, 3.0, 2.0, 1.0};
     while (state.KeepRunning()) {
-        inSort.sort(&a);
+        Insort.sort(&a);
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:69]: InSort

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 8]
```diff
@@ -43,28 +45,77 @@ static void BFSortDouble(benchmark::State& state) {
...
     }
 }
 
-static void InSortDouble(benchmark::State& state) {
-    llama::InsertionSort<double> inSort = llama::InsertionSort<double>();
+static void InsortDouble(benchmark::State& state) {
+    llama::InsertionSort<double> Insort = llama::InsertionSort<double>();
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:66]: InSort

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 9]
```diff
@@ -43,28 +45,77 @@ static void BFSortDouble(benchmark::State& state) {
...
     std::vector<double> a = {5.0, 4.0, 3.0, 2.0, 1.0};
     while (state.KeepRunning()) {
-        inSort.sort(&a);
+        Insort.sort(&a);
+    }
+}
+// SelctionSort
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:72]: Must be a sentence.

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 10]
```diff
@@ -43,28 +45,77 @@ static void BFSortDouble(benchmark::State& state) {
...
+
+static void SeSortDouble(benchmark::State& state) {
+    llama::SelectionSort<double> SeSort = llama::SelectionSort<double>();
+    std::vector<double> a = {5.0, 4.0, 3.0, 2.0, 1.0};
+    while (state.KeepRunning()) {
+        SeSort.sort(&a);
+    }
+}
+
+// BubbleSort
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:97]: Must be a sentence.

File: llama_cpp/example/sort_test/sort_test.cpp [hunk 11]
```diff
@@ -36,7 +38,7 @@ int main() {
 	std::vector<double> c;
 	std::vector<CustomisedClass> d;
 
-	// Original Data
+	// Set Original Data
```
Comment (0xsuu) [inline llama_cpp/example/sort_test/sort_test.cpp:41]: Full stop.

File: llama_cpp/example/sort_test/sort_test.cpp [hunk 12]
```diff
@@ -51,7 +53,7 @@ int main() {
 		std::cout << std::string(d[i]) << " " << std::endl;
 	}
 
-	// BruteForceSort
+	// BruteForceSort test
```
Comment (0xsuu) [inline llama_cpp/example/sort_test/sort_test.cpp:56]: Must be a sentence.

File: llama_cpp/example/sort_test/sort_test.cpp [hunk 13]
```diff
@@ -71,7 +73,7 @@ int main() {
 		std::cout << std::string(d[i]) << " " << std::endl;
 	}
 
-	// InsertionSort
+	// InsertionSort test
```
Comment (0xsuu) [inline llama_cpp/example/sort_test/sort_test.cpp:76]: Must be a sentence.

File: llama_cpp/src/Sort/SelectionSort.h (hunk 3 above)
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:27]: Still not fixed.

File: llama_cpp/src/Sort/SelectionSort.h (hunk 1 above)
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:22]: Still not fixed.

File: llama_cpp/src/Sort/SelectionSort.h (hunk 2 above)
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:21]: Still not fixed.

**Comment by 0xsuu** [comment]
Resolve conflicts please.

File: llama_cpp/src/Sort/SelectionSort.h [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+			for (int i = 0; i < reformedArray->size(); i++) {
+				SortObject<T> key = reformedArray->at(i);
+				int flag = i;
+				// Select the min value
+				for (int j = i; j < reformedArray->size(); j++){
+					if (key > reformedArray->at(j)){
+						flag = j;
+					}
+				}
+				// Swap between flag and the i if they are different
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:26]: !fix Add full stop.

**Pull request merged**